app/models/energy/scheduleruleset.py

```diff
@@ -0,0 +1,132 @@
+"""Schedule Ruleset Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError, UrlStr, ConstrainedStr
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+from app.models.energy.schedulebase import ScheduleNumericType, ScheduleUnitType
+from app.models.common.datetime import Date, Time
+
+
+class ScheduleTypeLimits(BaseModel):
+    """Specifies the data types and limits for values contained in schedules."""
+
+    type: Enum('ScheduleTypeLimits', {'type': 'ScheduleTypeLimits'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    lower_limit_value: float = Schema(
+        0,
+        description='Lower limit for the schedule type is entered.'
+    )
+
+    upper_limit_value: float = Schema(
+        1,
+        description='Upper limit for the schedule type is entered.'
+    )
+
+    numeric_type: ScheduleNumericType
+
+    unit_type: ScheduleUnitType
+
+
+class DayValue(BaseModel):
+    """Values for daily schedule"""
+
+    time: Time
+
+    value_until_time: float
+
+
+class ScheduleDay(BaseModel):
```

Comment:
This could be a ScheduleFixed interval with 24 values but I think this way is better (it is more compact and restricts to 24 values only).